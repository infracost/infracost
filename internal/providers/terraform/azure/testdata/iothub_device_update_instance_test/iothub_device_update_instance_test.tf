provider "azurerm" {
  skip_provider_registration = true
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_iothub_device_update_account" "example_account" {
  name                = "example-update-account"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_iothub" "iothub_single" {
  name                = "example-iothub"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  sku {
    name     = "S1"
    capacity = 1
  }

  tags = {
    purpose = "testing"
  }
}

resource "azurerm_iothub_device_update_instance" "standard_instance" {
  name                     = "example-update-instance"
  device_update_account_id = azurerm_iothub_device_update_account.example_account.id
  iothub_id                = azurerm_iothub.iothub_single.id
}
