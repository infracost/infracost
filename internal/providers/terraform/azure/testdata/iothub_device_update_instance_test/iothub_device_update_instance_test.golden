 Name                                                       Monthly Qty  Unit     Monthly Cost  
                                                                                               
 azurerm_iothub_device_update_instance.standard_instance           500  device         $50.00  
   └─ Device update devices                                        500  device         $50.00  

 OVERALL TOTAL                                                                         $50.00  

*Usage costs can be estimated by updating Infracost Cloud settings, see docs for other options.

──────────────────────────────────
1 cloud resource was detected:
∙ 1 was estimated

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━┓
┃ Project                                                       ┃ Baseline cost ┃ Usage cost* ┃ Total cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━╋━━━━━━━━━━━━┫
┃ iothub_device_update_instance_test                            ┃        $50.00 ┃           - ┃     $50.00 ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━┻━━━━━━━━━━━━┛
